slsim/Deflectors/DeflectorTypes/nfw_cluster.py

```diff
@@ -0,0 +1,66 @@
+from slsim.Deflectors.DeflectorTypes.deflector_base import DeflectorBase
+from slsim.Deflectors.velocity_dispersion import vel_disp_nfw_aperture
+from lenstronomy.Cosmo.lens_cosmo import LensCosmo
+
+
+class NFWCluster(DeflectorBase):
+    """Class of a NFW halo lens model with subhalos. Each subhalo is a Deflector
+    instance with its own mass and light.
+
+    required quantities in dictionary:
+    - 'halo_mass': halo mass in physical M_sol
+    - 'concentration': halo concentration
+    - 'e1_mass': eccentricity of NFW profile
+    - 'e2_mass': eccentricity of NFW profile
+    - 'z': redshift of deflector
+    """
+
+    def __init__(self, deflector_dict, subhalos_list):
+        super(NFWCluster, self).__init__(deflector_dict)
+        self._subhalos = subhalos_list
+
+    def velocity_dispersion(self, cosmo=None):
+        """Velocity dispersion of deflector. Simplified assumptions on anisotropy and
+        averaged over the characteristic radius.
+
+        :param cosmo: cosmology
+        :type cosmo: ~astropy.cosmology class
+        :return: velocity dispersion [km/s]
+        """
+        # convert radian to arc seconds
+        lens_cosmo = LensCosmo(z_lens=self.redshift, z_source=10, cosmo=cosmo)
+
+        m_halo, c_halo = self.halo_properties
+        rs_arcsec, _ = lens_cosmo.nfw_physical2angle(m_halo, c_halo)
+        vel_disp = vel_disp_nfw_aperture(
```

Comment:
Hi @sibirrer, indeed the velocity dispersion is the one of the dark matter particles without any light tracer. This is because the velocity dispersion of the dark matter halo of groups or clusters cannot be measured in the same way as galaxies. It is usually measured by the spread in the redshift of the subhalos within a radius and assuming the radial profile for the velocity dispersion from Łokas & Mamon 2001. So it should be consistent with the measured velocity dispersion with some considerations on the aperture radius.